src/main.py

```diff
@@ -144,18 +144,11 @@ def display_channel_to_article_type(display_channel_list):
 }
 
 def related_article_to_related_articles(related_article_list):
-    related_articles = []
-    if related_article_list:
-        for related in related_article_list:
-            try:
-                doi = related["xlink_href"]
-            except KeyError:
-                continue
-            if doi:
-                related_articles.append(doi.split('.')[-1])
-    if len(related_articles) <= 0:
-        return None
-    return related_articles
+    # ll: [{'xlink_href': u'10.7554/eLife.09561', 'related_article_type': u'article-reference', 'ext_link_type': u'doi'}]
+    def et(struct):
+        return struct.get('xlink_href', '').rsplit('.', 1)[-1] or None
+    # ll: ['09561'] or None
+    return filter(None, map(et, related_article_list)) or None
```

Comment:
preserving previous behaviour so tests continue to pass. I probably will change it to `[]` once I get stuck into it.